applications/GeoMechanicsApplication/custom_conditions/U_Pw_normal_lysmer_absorbing_condition.cpp

```diff
@@ -457,63 +462,82 @@ GetVariables(
     rVariables.virtual_thickness = this->GetValue(VIRTUAL_THICKNESS);
 }
 
+template< unsigned int TDim, unsigned int TNumNodes >
+void UPwLysmerAbsorbingCondition<TDim, TNumNodes>::
+CalculateAndAddRHS(VectorType& rRightHandSideVector, const MatrixType& rStiffnessMatrix)
+{
+
+    if (rRightHandSideVector.size() != CONDITION_SIZE)
+        rRightHandSideVector.resize(CONDITION_SIZE, false);
+    noalias(rRightHandSideVector) = ZeroVector(CONDITION_SIZE);
+
+    Vector displacements_vector = ZeroVector(CONDITION_SIZE);
+    this->GetValuesVector(displacements_vector, 0);
+
+    rRightHandSideVector -= prod(rStiffnessMatrix, displacements_vector);
+}
+
 
 template< unsigned int TDim, unsigned int TNumNodes >
 void UPwLysmerAbsorbingCondition<TDim, TNumNodes>::
-CalculateAndAddLHS(MatrixType& rLeftHandSideMatrix,
-    NormalLysmerAbsorbingVariables& rVariables)
+AddLHS(MatrixType& rLeftHandSideMatrix, const BoundedMatrix<double, N_DOF, N_DOF>& rUMatrix)
 {
-    //Adding contribution to left hand side
+	// assemble left hand side vector
+    if (rLeftHandSideMatrix.size1() != CONDITION_SIZE)
+        rLeftHandSideMatrix.resize(CONDITION_SIZE, CONDITION_SIZE, false);
+
+    noalias(rLeftHandSideMatrix) = ZeroMatrix(CONDITION_SIZE, CONDITION_SIZE);
```

Comment:
Also here I suspect that we might be able to simplify this to:
```suggestion
    rLeftHandSideMatrix = ZeroMatrix(CONDITION_SIZE, CONDITION_SIZE);
```